core/utils/pqueue.c

```diff
@@ -473,3 +473,63 @@ void print_event(void *event) {
     LF_PRINT_DEBUG("time: " PRINTF_TIME ", trigger: %p, token: %p",
             e->time, e->trigger, e->token);
 }
+
+
+// ********** NDT Priority Queue Support Start
+/**
+ * Return whether the tags contained by the first and second argument 
+ * are given in reverse order.
+*/
+int tag_in_reverse_order(pqueue_pri_t thiz, pqueue_pri_t that) {
+    return lf_tag_compare(((ndt_node*) thiz)->tag, ((ndt_node*) that)->tag) == 1;
+}
+
+/**
+ * Return whether or not the tags contained by given pointers are identical.
+ */
+int tag_matches(void* next, void* curr) {
+    return lf_tag_compare(((ndt_node*) next)->tag, ((ndt_node*) curr)->tag);
```

Comment:
This line does not look right.  It will return 0 (false) if the tags are identical.

Reply:
Oh, I missed it. It's fixed in the commit [3faa188](https://github.com/lf-lang/reactor-c/pull/176/commits/3faa18840cce38d89d8d833ba8ce03ba097b6606). Thank you for letting me know, @edwardalee!